pkg/errors/errors.go

```diff
@@ -4,9 +4,19 @@ import (
 	"fmt"
 	"strings"
 
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/klog"
 )
 
+// kubeobj is close to an union of Object and Type, as there's no interface in
+// apimachinery that has them both together, and it's used in a few places here
+// to get k8s objects as a parameter.
+type kubeobj interface {
```

Comment:
Should we move it outside errors package and make it public maybe? I see some potential in a broader usage of this interface.

Reply:
Yes, we probably should, I wanted to reach out for it once or twice. Any ideas in which package this should live?